### PR TITLE
feat(sdk)!: enable open telemetry by default.

### PR DESCRIPTION
**Pull request by vkubiv**

### CODE REVIEW

**Comment by rolsonquadras** [comment]
Can we update the doc to add new TraceID field to error handling https://github.com/trustbloc/wallet-sdk/blob/main/cmd/wallet-sdk-gomobile/docs/usage.md#error-handling section?

**Review by birtony — APPROVED** [review]

File: cmd/wallet-sdk-gomobile/openid4vp/interaction.go [hunk 1]
```diff
@@ -39,15 +40,29 @@ type Interaction struct {
 	goAPIOpenID4VP   goAPIOpenID4VP
 	didResolver      api.DIDResolver
 	inquirer         *credential.Inquirer
+	oTel             *otel.Trace
 }
 
 // NewInteraction creates a new OpenID4VP Interaction.
 // The methods defined on this object are used to help guide the calling code through the OpenID4CI flow.
-func NewInteraction(args *Args, opts *Opts) *Interaction {
+func NewInteraction(args *Args, opts *Opts) (*Interaction, error) { //nolint:funlen
```
Comment (DRK3) [inline cmd/wallet-sdk-gomobile/openid4vp/interaction.go:48]: Need to update `usage.md` to show the error being passed in in the Swift examples (you can follow the pattern from the other Swift examples)

Also this is a breaking change, so this PR should have an `!` in it (i.e. `feat(sdk)!`)

**Review by DRK3 — COMMENTED** [review]
Left one comment, otherwise looks good!

File: cmd/wallet-sdk-gomobile/docs/usage.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ Errors have three fields:
 * Code: A short alphanumeric code that is used to broadly group various related errors.
 This contains the Category descriptor.
 * Details: Lower-level details about the precise cause of the error.
+* TraceID: An id of Open Telemetry root trace. Can be used to trace API calls.
```
Comment (DRK3) [inline cmd/wallet-sdk-gomobile/docs/usage.md:40]:
```suggestion
* TraceID: ID of Open Telemetry root trace. Can be used to trace API calls. Only present on certain APIs.
```

^Also perhaps which APIs have this?

File: cmd/wallet-sdk-gomobile/openid4ci/opts.go [hunk 3]
```diff
@@ -63,3 +64,8 @@ func (o *Opts) SetActivityLogger(activityLogger api.ActivityLogger) {
 func (o *Opts) SetMetricsLogger(metricsLogger api.MetricsLogger) {
 	o.metricsLogger = metricsLogger
 }
+
+// DisableOpenTelemetry disables sending of open telemetry header.
+func (o *Opts) DisableOpenTelemetry() {
```
Comment (DRK3) [inline cmd/wallet-sdk-gomobile/openid4ci/opts.go:69]: Can you update this to allow for Opts chaining so it matches my open PR (#383)?

**Pull request merged**